app/controllers/concerns/foreman/controller/smart_proxy_auth.rb

```diff
@@ -18,17 +18,11 @@ def add_smart_proxy_filters(actions, options = {})
       self.smart_proxy_filter_actions ||= []
       self.smart_proxy_filter_actions.push(*actions)
 
-      prepend SmartProxyRequireSsl
-    end
-  end
-
-  module SmartProxyRequireSsl
-    def require_ssl?
-      if [self.smart_proxy_filter_actions].flatten.map(&:to_s).include?(action_name)
-        false
```

Comment:
I still wonder about this. Did this ever work correctly? It feels to me like this should have been `true`. With that in mind, I also think the replacement code you have may also be incorrect. So let's first figure this out because I suspect we may be able to just drop it altogether.

Reply:
it is about the message. The message will not be shown if this would be true as that would redirect before we get to it.
See the test I've added. Lets discuss if we need this whole hacky solution to get that message in logs :shrug: